pkg/controller/devicetainteviction/device_taint_eviction_test.go

```diff
@@ -494,6 +498,24 @@ func TestHandlers(t *testing.T) {
 			// At the moment, the code reliably cancels right away.
 			wantEvents: []*v1.Event{cancelPodEviction},
 		},
+		"evict-pod-after-scheduling": {
+			initialState: state{
+				pods:            []*v1.Pod{unscheduledPodWithClaimName},
+				slices:          []*resourceapi.ResourceSlice{sliceTainted, slice2},
+				allocatedClaims: []allocatedClaim{{ResourceClaim: inUseClaim, evictionTime: &taintTime}},
+				evicting:        []evictAt{{newObject(podWithClaimName), taintTime.Time}},
```

Comment:
Is the idea with this test that the pod should not be queued for eviction at first, but then is after the pod is updated? I see this test pass even without the NodeName check otherwise.

Reply:
Oh, right, I need to remove the initial evicting state. Then it fails without the `oldPod.Spec.NodeName == newPod.Spec.NodeName`.

Fixed.